core/rawdb/chain_iterator.go

```diff
@@ -50,15 +50,17 @@ func InitDatabaseFromFreezer(db ethdb.Database) {
 		// freezerdb return N items (e.g up to 1000 items per go)
 		// That would require an API change in Ancients though
 		if h, err := db.Ancient(freezerHashTable, i); err != nil {
-			log.Crit("Failed to init database from freezer", "err", err)
+			log.Error("Failed to init database from freezer", "err", err)
+			return
```

Comment:
It's probably problematic. If we plug in an external ancient to an empty Geth, we hold the assumption that after calling `InitDatabaseFromFreezer`, all canonical hashes will be registered and the `HeadHeaderHash` and `HeadFastBlockHash` will be updated. 

But if we abort without returning any error, it breaks the assumption.

Reply:
Yeah maybe we should leave the Crit for `InitDatabaseFromFreezer`. But we should check what happens on ctrl-c during this operation

Reply:
fixed